src/GitVersionCore/VersionCalculation/NextVersionCalculator.cs

```diff
@@ -62,11 +63,20 @@ public SemanticVersion FindVersion(GitVersionContext context)
                 UpdatePreReleaseTag(context, semver, baseVersion.BranchNameOverride);
             }
 
-
             if (taggedSemanticVersion != null)
             {
-                // set the commit count on the tagged ver
-                taggedSemanticVersion.BuildMetaData.CommitsSinceVersionSource = semver.BuildMetaData.CommitsSinceVersionSource;
+                // replace calculated version with tagged version only if tagged version greater or equal to calculated version
```

Comment:
Do you think we can use semver > taggedSemanticVersion instead? as the SemanticVersion implements IComparable

Reply:
Full semver compare leads to failure in some tests, eg. DevelopScenarios.CanHandleContinuousDelivery -> 1.1.0-alpha.8+1 instead of 1.1.0-alpha.7

At first sight it seems GitVersion is calculating wrong.